config/v1/types_infrastructure.go

```diff
@@ -1417,6 +1420,85 @@ type NutanixPlatformSpec struct {
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z_.-@/]+$`
+	Name string `json:"name"`
+
+	// prismCentralEndpoint holds the endpoint (address, port) to access the Nutanix Prism Central.
+	// If this is not configured, it means this failure domain uses the default prism_central
+	// endpoint as configured in the nutanix platformSpec.
+	// +optional
+	PrismCentralEndpoint *NutanixPrismEndpoint `json:"prismCentralEndpoint,omitempty"`
+
+	// prismElementEndpoint holds the endpoint address and port to access the Nutanix Prism Element.
+	// When a cluster-wide proxy is installed, by default, this endpoint will be accessed via the proxy.
+	// Should you wish for communication with this endpoint not to be proxied, please add the endpoint to the
+	// proxy spec.noProxy list.
+	// +optional
+	PrismElementEndpoint *NutanixPrismElementEndpoint `json:"prismElementEndpoint,omitempty"`
+
+	// cluster is to identify the cluster (the Prism Element under management of the Prism Central),
+	// in which the Machine's VM will be created. The cluster identifier (uuid or name) can be obtained
+	// from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnets holds a list of identifiers (one or more) of the cluster's network subnets
+	// for the Machine's VM to connect to. The subnet identifiers (uuid or name) can be
+	// obtained from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=atomic
+	Subnets []NutanixResourceIdentifier `json:"subnets"`
+
+	// credentialsSecret is a local reference to a secret that contains the
+	// credentials data to access the prism_central. If this is not configured,
+	// it means this failure domain uses the default credentials secret name.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+}
+
+// NutanixIdentifierType is an enumeration of different resource identifier types.
+type NutanixIdentifierType string
+
+const (
+	// NutanixIdentifierUUID is a resource identifier identifying the object by UUID.
+	NutanixIdentifierUUID NutanixIdentifierType = "uuid"
+
+	// NutanixIdentifierName is a resource identifier identifying the object by Name.
+	NutanixIdentifierName NutanixIdentifierType = "name"
+)
+
+// NutanixResourceIdentifier holds the identity of a Nutanix PC resource (cluster, image, subnet, etc.)
+// +union
+type NutanixResourceIdentifier struct {
+	// Type is the identifier type to use for this resource.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=uuid;name
+	Type NutanixIdentifierType `json:"type"`
+
+	// uuid is the UUID of the resource in the PC.
+	// +optional
+	UUID *string `json:"uuid,omitempty"`
+
+	// name is the resource name in the PC
+	// +optional
+	Name *string `json:"name,omitempty"`
```

Comment:
Can you expand on the godoc of these values?
Is `""` a valid value for either of these fields? If not, set them to not pointers

Reply:
Same as above mentioned, since the NutanixResourceIdentifier is used from the beginning (4.12), I think it is better not to change it to avoid any possible backward compatibility issue.

Reply:
Addressed.